### ISSUE TITLE
[RFE]: Offending line-logger

### ISSUE THREAD

**Issue by drwetter** [fail2ban/fail2ban#2972]
Hope I didn't miss anything...

#### Feature request type

Log the offending line maybe with ``loglevel=DEBUG`` in ``/var/log/fail2ban.log``

#### Description

I would like to see what regex succeeded or failed (false positive). It's difficult otherwise if you don't have a single regex in a filter defined.

It also would be useful for statistics.

(not talking about ``fail2ban-regex`` as a cli) 

PS: Thanks for the great tool!

**Comment by sebres** [comment]
It is possible to log which RE is matching using heavydebug log-level (just it is too wordy and not recommended to use regularly).

> I would like to see what regex succeeded or failed (false positive)

If your intention is just to find responsible REs matched by some IPs (e. g. searching for false positives), you can simply use `fail2ban-regex` over an excerpt for grep by IP:
```bash
grep -wF 192.0.2.1 /var/log/auth.log > /tmp/log
fail2ban-regex -vv /tmp/log sshd
```
Additionally to matching REs you'd also see the time (and IP) when this RE gets matched.

As regards the possibility to output statistic, I have experimental branch where every REs have a matched count (used to speedup parser by reordering of REs, so most used at first). This can be indeed extended with a statistic facilities.

**Comment by drwetter** [comment]
Hi @sebres ,

thanks for the instructions. But I was more hinting that this would be a great builtin feature, especially as fail2ban has this information already.
